### PR TITLE
MPI - Actualiza funciones y rutas

### PR DESCRIPTION
**Pull request by negro89**
### Requerimiento
https://proyectos.andes.gob.ar/browse/MPI-308

### Funcionalidad desarrollada 

1. Actualiza funciones y rutas

### UserStories llegó a completarse

- [ ] Si
- [X] No

### Requiere actualizaciones en la base de datos

- [ ] Si
- [X] No

### CODE REVIEW

**Comment by liquid36** [comment]
USUARIO: lmlagos  
BUILD NUMBER: [2920](http://condorpiedra.andes.gob.ar:85/2920/mochawesome.html)
CYPRESS RUN: [2053](https://dashboard.cypress.io/projects/xr7gft/runs/2053/specs)
TEST START: 2020-12-17T14:36:51.847Z
TOTAL: 353
SUCCESS: 331
FAIL: 22
SKIPPED: 0

**Comment by liquid36** [comment]
USUARIO: lmlagos  
BUILD NUMBER: [2933](http://condorpiedra.andes.gob.ar:85/2933/mochawesome.html)
CYPRESS RUN: [2066](https://dashboard.cypress.io/projects/xr7gft/runs/2066/specs)
TEST START: 2020-12-18T13:16:30.049Z
TOTAL: 353
SUCCESS: 341
FAIL: 12
SKIPPED: 0

**Comment by liquid36** [comment]
USUARIO: lmlagos  
BUILD NUMBER: [2936](http://condorpiedra.andes.gob.ar:85/2936/mochawesome.html)
CYPRESS RUN: [2069](https://dashboard.cypress.io/projects/xr7gft/runs/2069/specs)
TEST START: 2020-12-18T15:50:24.838Z
TOTAL: 353
SUCCESS: 352
FAIL: 1
SKIPPED: 0

**Comment by liquid36** [comment]
USUARIO: lmlagos  
BUILD NUMBER: [2939](http://condorpiedra.andes.gob.ar:85/2939/mochawesome.html)
CYPRESS RUN: [2072](https://dashboard.cypress.io/projects/xr7gft/runs/2072/specs)
TEST START: 2020-12-18T16:33:01.340Z
TOTAL: 356
SUCCESS: 356
FAIL: 0
SKIPPED: 0

File: core-v2/mpi/paciente/paciente.routes.ts [hunk 1]
```diff
@@ -142,14 +143,17 @@ export const get = async (req: Request, res: Response) => {
...
     const base64RegExp = /data:([a-zA-Z0-9]+\/[a-zA-Z0-9-.+]+).*,(.*)/;
 
     if (!(mongoose.Types.ObjectId.isValid(req.params.id))) {
         return next(404);
     }
-    const pacienteBuscado: any = await Paciente.findById(req.params.id);
+    const pacienteBuscado: any = await Paciente.findById(req.params.id, '+foto');
     if (pacienteBuscado) {
-        if (!pacienteBuscado.foto) {
+        if (!pacienteBuscado.fotoId || !pacienteBuscado.foto || pacienteBuscado.foto == null) {
```
Comment (sychus) [inline core-v2/mpi/paciente/paciente.routes.ts:156]: no alcanza con: !pacienteBuscado.fotoId || !pacienteBuscado.foto ??

**Review by sychus — CHANGES_REQUESTED** [review]
revisar detalle

**Review by palita1991 — APPROVED** [review]

File: core-v2/mpi/paciente/paciente.routes.ts [hunk 2]
```diff
@@ -121,14 +116,15 @@ export const find = async (req: Request, res: Response) => {
...
         const pacientes = await multimatch(req.query.search, conditions, options);
-        res.json(pacientes);
+        return res.json(pacientes);
     } else {
         const conditions = req.query;
         const pacientes = await PacienteCtr.search(conditions, options, req);
-        res.json(pacientes);
+        return res.json(pacientes);
     }
 };
```
Comment (sychus) [inline core-v2/mpi/paciente/paciente.routes.ts:129]: No debería tener un try catch por si alguna de las funciones que tiene falla?

File: core-v2/mpi/paciente/paciente.routes.ts [hunk 3]
```diff
@@ -233,7 +194,7 @@ export const post = async (req: Request, res: Response) => {
         paciente.identificadores = valor ? [{ entidad: 'RENAPER', valor }] : null;
     }
     const pacienteCreado = await PacienteCtr.create(paciente, req);
-    res.json(pacienteCreado);
+    return res.json(pacienteCreado);
 };
```
Comment (sychus) [inline core-v2/mpi/paciente/paciente.routes.ts:198]: idem anterior (try catch)

**Review by sychus — COMMENTED** [review]
tema: try catch

**Comment by liquid36** [comment]
USUARIO: cramos  
BUILD NUMBER: [3120](http://condorpiedra.andes.gob.ar:85/3120/mochawesome.html)
CYPRESS RUN: [2246](https://dashboard.cypress.io/projects/xr7gft/runs/2246/specs)
TEST START: 2021-01-26T13:44:03.224Z
TOTAL: 327
SUCCESS: 315
FAIL: 12
SKIPPED: 0
:warning::warning::warning::warning:
Hay archivos que no corrieron. Vuelva a intentar.

**Comment by liquid36** [comment]
USUARIO: cramos  
BUILD NUMBER: [3121](http://condorpiedra.andes.gob.ar:85/3121/mochawesome.html)
CYPRESS RUN: [2247](https://dashboard.cypress.io/projects/xr7gft/runs/2247/specs)
TEST START: 2021-01-26T14:10:15.255Z
TOTAL: 354
SUCCESS: 353
FAIL: 1
SKIPPED: 0

**Comment by liquid36** [comment]
USUARIO: cramos  
BUILD NUMBER: [3122](http://condorpiedra.andes.gob.ar:85/3122/mochawesome.html)
CYPRESS RUN: [2248](https://dashboard.cypress.io/projects/xr7gft/runs/2248/specs)
TEST START: 2021-01-26T14:18:26.738Z
TOTAL: 351
SUCCESS: 351
FAIL: 0
SKIPPED: 0

**Comment by liquid36** [comment]
USUARIO: nhuenchuman  
BUILD NUMBER: [3124](http://condorpiedra.andes.gob.ar:85/3124/mochawesome.html)
CYPRESS RUN: [2250](https://dashboard.cypress.io/projects/xr7gft/runs/2250/specs)
TEST START: 2021-01-26T15:10:23.640Z
TOTAL: 354
SUCCESS: 353
FAIL: 1
SKIPPED: 0

**Comment by liquid36** [comment]
USUARIO: nhuenchuman  
BUILD NUMBER: [3125](http://condorpiedra.andes.gob.ar:85/3125/mochawesome.html)
CYPRESS RUN: [2251](https://dashboard.cypress.io/projects/xr7gft/runs/2251/specs)
TEST START: 2021-01-26T15:47:38.565Z
TOTAL: 354
SUCCESS: 349
FAIL: 5
SKIPPED: 0

**Comment by liquid36** [comment]
USUARIO: cramos  
BUILD NUMBER: [3126](http://condorpiedra.andes.gob.ar:85/3126/mochawesome.html)
CYPRESS RUN: [2252](https://dashboard.cypress.io/projects/xr7gft/runs/2252/specs)
TEST START: 2021-01-26T15:57:32.929Z
TOTAL: 354
SUCCESS: 350
FAIL: 4
SKIPPED: 0

**Comment by liquid36** [comment]
USUARIO: nhuenchuman  
BUILD NUMBER: [3127](http://condorpiedra.andes.gob.ar:85/3127/mochawesome.html)
CYPRESS RUN: [2253](https://dashboard.cypress.io/projects/xr7gft/runs/2253/specs)
TEST START: 2021-01-26T16:51:07.859Z
TOTAL: 354
SUCCESS: 351
FAIL: 3
SKIPPED: 0

**Comment by liquid36** [comment]
USUARIO: nhuenchuman  
BUILD NUMBER: [3128](http://condorpiedra.andes.gob.ar:85/3128/mochawesome.html)
CYPRESS RUN: [2254](https://dashboard.cypress.io/projects/xr7gft/runs/2254/specs)
TEST START: 2021-01-26T17:32:36.111Z
TOTAL: 314
SUCCESS: 302
FAIL: 12
SKIPPED: 0
:warning::warning::warning::warning:
Hay archivos que no corrieron. Vuelva a intentar.

**Comment by liquid36** [comment]
USUARIO: nhuenchuman  
BUILD NUMBER: [3129](http://condorpiedra.andes.gob.ar:85/3129/mochawesome.html)
CYPRESS RUN: [2255](https://dashboard.cypress.io/projects/xr7gft/runs/2255/specs)
TEST START: 2021-01-26T18:11:56.330Z
TOTAL: 354
SUCCESS: 351
FAIL: 3
SKIPPED: 0

**Comment by liquid36** [comment]
USUARIO: nhuenchuman  
BUILD NUMBER: [3130](http://condorpiedra.andes.gob.ar:85/3130/mochawesome.html)
CYPRESS RUN: [2256](https://dashboard.cypress.io/projects/xr7gft/runs/2256/specs)
TEST START: 2021-01-26T18:25:01.782Z
TOTAL: 354
SUCCESS: 351
FAIL: 3
SKIPPED: 0

**Comment by liquid36** [comment]
USUARIO: nhuenchuman  
BUILD NUMBER: [3131](http://condorpiedra.andes.gob.ar:85/3131/mochawesome.html)
CYPRESS RUN: [2257](https://dashboard.cypress.io/projects/xr7gft/runs/2257/specs)
TEST START: 2021-01-26T18:42:03.677Z
TOTAL: 351
SUCCESS: 348
FAIL: 3
SKIPPED: 0

**Comment by liquid36** [comment]
USUARIO: cramos  
BUILD NUMBER: [3133](http://condorpiedra.andes.gob.ar:85/3133/mochawesome.html)
CYPRESS RUN: [2259](https://dashboard.cypress.io/projects/xr7gft/runs/2259/specs)
TEST START: 2021-01-26T21:12:52.065Z
TOTAL: 354
SUCCESS: 353
FAIL: 1
SKIPPED: 0

**Comment by liquid36** [comment]
USUARIO: nhuenchuman  
BUILD NUMBER: [3135](http://condorpiedra.andes.gob.ar:85/3135/mochawesome.html)
CYPRESS RUN: [2261](https://dashboard.cypress.io/projects/xr7gft/runs/2261/specs)
TEST START: 2021-01-27T10:32:24.622Z
TOTAL: 354
SUCCESS: 354
FAIL: 0
SKIPPED: 0

**Comment by liquid36** [comment]
USUARIO: cramos  
BUILD NUMBER: [3136](http://condorpiedra.andes.gob.ar:85/3136/mochawesome.html)
CYPRESS RUN: [2262](https://dashboard.cypress.io/projects/xr7gft/runs/2262/specs)
TEST START: 2021-01-27T12:21:32.555Z
TOTAL: 313
SUCCESS: 293
FAIL: 20
SKIPPED: 0
:warning::warning::warning::warning:
Hay archivos que no corrieron. Vuelva a intentar.

**Comment by liquid36** [comment]
USUARIO: cramos  
BUILD NUMBER: [3137](http://condorpiedra.andes.gob.ar:85/3137/mochawesome.html)
CYPRESS RUN: [2263](https://dashboard.cypress.io/projects/xr7gft/runs/2263/specs)
TEST START: 2021-01-27T12:29:55.719Z
TOTAL: 354
SUCCESS: 354
FAIL: 0
SKIPPED: 0

**Review by condorpiedra — APPROVED** [review]

**Pull request merged**